src/commands/flow/internal/spec/saveToCsv.spec.ts

```diff
@@ -0,0 +1,102 @@
+import path from 'path'
+import { deleteFile, generateTimestamp, readFile } from '@sasjs/utils'
+import { saveToCsv } from '..'
+import csvColumns from '../csvColumns'
+
+describe('saveToCsv', () => {
+  let csvFilePath: string
+
+  afterEach(async () => {
+    await deleteFile(csvFilePath)
+  })
+
+  it('should write to csv', async () => {
+    csvFilePath = path.join(__dirname, `cli-tests-csv-${generateTimestamp()}`)
+
+    await saveToCsv(
+      csvFilePath,
+      [
+        'someFlowName',
+        ['flow1', 'flow2'].join(' | '),
+        'location',
+        'success',
+        undefined,
+        undefined
+      ],
+      Object.values(csvColumns)
+    )
+
+    const csvContent = await readFile(csvFilePath)
+
+    expect(csvContent).toEqual(
+      expect.stringContaining('1,someFlowName,flow1 | flow2,location,success,,')
+    )
+  })
+
+  it('should write to csv - asynchronous at same time', async () => {
+    csvFilePath = path.join(__dirname, `cli-tests-csv-${generateTimestamp()}`)
+
+    await Promise.all(
+      [1, 2, 3].map(async (_, i) => {
+        await new Promise((r) => setTimeout(r, 1000 * (i - 1) * (i - 1)))
```

Comment:
the `st index will be `0`, so `0 - 1` will result in a negative number

Reply:
for `i`;
- at 0, it will get (-1) * (-1) = 1
- at 1, it will get 0 *0 = 0
- at 2, it will get 1 * 1 = 1

So, this test case is essential for testing 1st and 3rd promise resolves at same time after 2nd.

Reply:
aha, okay